all-actors/src/main/java/org/sunbird/serviceimpl/CertsServiceImpl.java

```diff
@@ -0,0 +1,177 @@
+package org.sunbird.serviceimpl;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.mashape.unirest.http.HttpResponse;
+import com.mashape.unirest.http.JsonNode;
+import org.apache.commons.collections.MapUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.log4j.Logger;
+import org.sunbird.BaseException;
+import org.sunbird.CertVars;
+import org.sunbird.JsonKeys;
+import org.sunbird.builders.Certificate;
+import org.sunbird.builders.Course;
+import org.sunbird.message.IResponseMessage;
+import org.sunbird.message.ResponseCode;
+import org.sunbird.request.Request;
+import org.sunbird.response.Response;
+import org.sunbird.service.ICertService;
+import org.sunbird.utilities.CertificateUtil;
+
+import java.text.MessageFormat;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Future;
+
+public class CertsServiceImpl implements ICertService {
+    static Logger logger = Logger.getLogger(CertsServiceImpl.class);
+    private static ObjectMapper requestMapper = new ObjectMapper();
+
+    @Override
+    public String add(Request request) throws BaseException {
+        Map<String, Object> certAddReqMap = new HashMap<>();
+        certAddReqMap = request.getRequest();
+        assureUniqueCertId((String) certAddReqMap.get(JsonKeys.ID));
+        processRecord(certAddReqMap);
+        logger.info("CertsServiceImpl:add:record successfully processed with request:"+certAddReqMap);
+        return (String)certAddReqMap.get(JsonKeys.ID);
+    }
+
+    private void assureUniqueCertId(String certificatedId) throws BaseException {
+        if (CertificateUtil.isIdPresent(certificatedId)) {
+            logger.error(
+                    "CertificateActor:addCertificate:provided certificateId exists in record:" + certificatedId);
+            throw new BaseException(IResponseMessage.INVALID_REQUESTED_DATA, IResponseMessage.ID_ALREADY_EXISTS, ResponseCode.CLIENT_ERROR.getCode());
+        }
+        logger.info("CertificateActor:addCertificate:successfully certId not found in records creating new record");
+    }
+
+
+    private void processRecord(Map<String, Object> certReqAddMap){
+        Certificate certificate=getCertificate(certReqAddMap);
+        Map<String,Object>recordMap= requestMapper.convertValue(certificate,Map.class);
```

Comment:
why we are first converting map to Certificate object and then certificate object to map back.

Reply:
I personally think to build an object first from request and process it and then send in a form of pure sanitized data map to ES